activerecord-trilogy-adapter.gemspec

```
@@ -19,7 +19,7 @@ Gem::Specification.new do |spec|
   }
 
   spec.add_dependency "trilogy", ">= 2.1.1"
-  spec.add_dependency "activerecord", ">= 7.1.0.alpha"
+  spec.add_dependency "activerecord", "~> 7.1.0.alpha"
```

```suggestion
  spec.add_dependency "activerecord", "~> 7.1.x"
```

☝🏻 will allow all 7.1.0 prereleases as well as all of the 7.1.* series (and also not the 7.2.* series), which I think is what we want. 7.1.0.alpha would _only_ work on pre-7.1.0 prereleases... which is perhaps more correct, but probably not the intent here.

(On reflection, we arguably shouldn't have published a whole-numbered release with a dependency on the still-in-flux future 7.1 API, but 🤷🏻‍♂️)

---

I did try this and it doesn't seem to do what I thought it would,

```
Fetching gem metadata from https://rubygems.org/......
Resolving dependencies...
Bundler could not find compatible versions for gem "activerecord":
  In snapshot (Gemfile.lock):
    activerecord (= 7.1.0.alpha)

  In Gemfile:
    activerecord

    activerecord-trilogy-adapter was resolved to 2.0.0, which depends on
      activerecord (~> 7.1.x)

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
⋊> ~/c/activerecord-trilogy-adapter on allow-all-activerecord-7.1.0.alpha-versions ⨯ bundle update activerecord                                                                                                   08:17:32
Fetching https://github.com/rails/rails
Fetching gem metadata from https://rubygems.org/......
Resolving dependencies...
Bundler could not find compatible versions for gem "activerecord":
  In Gemfile:
    activerecord

    activerecord-trilogy-adapter was resolved to 2.0.0, which depends on
      activerecord (~> 7.1.x)

Could not find gem 'activerecord (~> 7.1.x)', which is required by gem 'activerecord-trilogy-adapter', in https://github.com/rails/rails (at main@2418939).

The source contains the following gems matching 'activerecord':
  * activerecord-7.1.0.alpha
 
```

---

How would you feel about:

```
spec.add_dependency "activerecord", "> 7.0"
```

It is probably a little too broad, but since there isn't an official 7.1 it is correct.

---

That includes all of 7.0.*.

And even while we're playing fast and loose with our compatibility claims on 7.1.*, I don't want to claim 7.2.* -- this isn't a "we can't be 100% sure of the future" situation, like all adapters, we depend on a variety of APIs that upstream absolutely does change between pretty much every release.

---

Argh, I forgot that Rubygems changed the definition of version constraints -- in older versions, 7.1.x _did_ match what we want, but no more. You can use 7.1.a instead.

---

The detail is that `1.2.x` used to come before all of `1.2.0.aaa` -- it's a pre-release of `1.2`, which comes before `1.2.0` -- but now `1.2.x` and `1.2.0.x` are treated as the same version.

In our case, `7.1.0.alpha.gxxx` comes before `7.1.0.alpha` (it's a prerelease of a prerelease). `7.1.x` previously came before all that [as a prerelease of the never-actual version "7.1"], but it's now effectively `7.1.0.x` [and `x` > `alpha`]. So instead we can use `7.1.a` aka `7.1.0.a`, because `a` < `alpha`, and therefore `a` precedes even a prerelease of `alpha`.

(So we're basically doing the same thing as `.x`, which was just a slightly more cute spelling of the same thing -- the `x` looked like it had semantic meaning [that it didn't], whereas the `a` is more clearly just a comparable string.)